### ISSUE TITLE
Feature request: CHK compression/protection without adding EUD trigger

### ISSUE THREAD

**Issue by Chromowolf** [armoha/euddraft#29]
Some SCMD-made maps need extended unit limit. Also I wanna prevent it from being modified by noob/average map makers. Traditional map protectors like SMLP, SMC are all outdated and cannot protect maps with "STRx". They also crack easily.
I know the elementary operation to protect a map is to do the CHK compression, i.e. remove the "ISOM", "TILE", location names, switch names and etc, which is exactly what euddraft does. Euddraft is by far the ONLY tool which can deal with maps using "STRx".
Could you add an option: Simply do the basic CHK compression, without adding EUD triggers, keeping the unit limit extended?

**Comment by armoha** [comment]
We don't do chk protection much and most technique are focused on runtime verification. Also I think this issue needs separate project, not in **eud**draft.

**Comment by Chromowolf** [comment]
> We don't do chk protection much

I don't actually need "much". What euddraft currently does is enough, Afaik, euddraft add EUD triggers to achieve runtime verifition, and in the very last step, do the chk compression (remove the "ISOM", "TILE", location names, and etc). I just wanna add an option (command line option like --no-eud-triggers) to let euddraft ONLY do the last step.

**Comment by armoha** [comment]
Removing unused section for SC is done in almost every protections. IMO euddraft map without freeze protection is pretty weak and literally not obfuscated or protected at all. Although SCMD2 can't open map but https://github.com/Buizz/UseMapEditor can and maybe Chkdraft can open too... Hence I keep inform euddraft without freeze is not a protection.

**Comment by Chromowolf** [comment]
>Although SCMD2 can't open map

90%+ map makers use only SCMD. Many of them just edit map for fun, and sometimes don't consider "copy right" and modify others works without giving credit to the original author. If SCMD cannot open the map, then they'll just give up, without trying to search for other tools. This group of people is what I called "noob/average" map makers, and also is what I wanna screen out.
> maybe Chkdraft can open too...

No. CHKdraft can't. I tried. It can only open some old protected maps.
>Removing unused section for SC is done in almost every protections.

The point is that, others tools are all outdated. Whenever there is "STRx" section in the map, the tool crashes.

>Hence I keep inform euddraft without freeze is not a protection.

Actually I don't want the real protection. All I want is just prevent those average map makers from opening the map in SCMD. If they have other tools which can open the map, then, well, let it go. Even though they can open it, there are no location names or switch names for their reference. So they couldn't modify anything corrensponding to their thoughts.

> but https://github.com/Buizz/UseMapEditor can

This is cool! Seems to be a new project. (Haven't got any release yet?)

Anyway I got your idea. Just in case you are interested in starting a new project for non-EUD map protection: All the existing non-EUD map protectors are outdated (cannot deal with "STRx") and the authors all seem to stop updating.

Thank you!

**Comment by armoha** [comment]
I actually had private work-in-progress protector project with new protection methods tailored for non-eud SC:R maps but abandoned that due to bug. ;<

Not publicly announced but new version of SMLP will be released soon or within a year maybe so I'd rather avoid duplicating work. Also I'm not sure whether non-eud maps are huge demand or pretty niche uses, or what extent do most target users want for protection.

Anyway I can't promise but maybe I could poke this again.

**Comment by armoha** [comment]
> If they have other tools which can open the map, then, well, let it go. 

Well, copying terrain from freeze protected maps is possible and takes place for several maps. IMHO things change when your map become popular.

> This is cool! Seems to be a new project. (Haven't got any release yet?)

Test version: https://cafe.naver.com/edac/99331
It is still quite WIP, can't edit terrain, not to mention translation.

**Comment by Chromowolf** [comment]
> new version of SMLP will be released soon or within a year

Sounds exciting. I'll wait for that. So far I'll just manually remove the unused sections using Hex Editor.
Thank you for the information of UseMapEditor. Looking forward to the official release.

Btw are you using any online chatting apps? It may be more convenient than github issues or emails, in case I have some SC questions to consult you. Some questions may be irrelevant to euddraft so I don't think it's proper to consult you in github issue. Like (hope you don't mind): 
I can't join the edac cafe. I registered a naver account, but many posts are opened only to edac cafe members. It says I must have an I-PIN or cellphone with my name. I don't have a "주민번호", or cellphone operated by "KT, SKT...". Any other way to join the cafe?

**Comment by armoha** [comment]
Discord is preferred to contact (`Armo#6637`)

**Comment by armoha** [comment]
http://blog.naver.com/fullhyuk/221462331342/
This blog post well describes how to sign up for foreigner without korean phone, but it maybe hard to follow up as it contains screenshot with korean.

**Comment by Chromowolf** [comment]
Thx a lot!
